components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/cairo-builtins.adoc

```diff
@@ -0,0 +1,26 @@
+[id="cairo-builtins"]
+= Cairo builtins
+
+_Builtins_ in Cairo are predefined optimized low-level execution units that the Cairo VM refers to in order to perform predefined computations that are expensive to perform in standard Cairo. Builtins enhance the functionality of the Cairo VM, enabling you to perform certain tasks, such as using the Poseidon hash, range-checks, or ECDSA signature verifications, more efficiently, using less gas.
+
+In contrast to CairoZero, where you needed to consciously write code to take advantage of builtin optimizations, in Cairo, you simply write code without doing anything special, and when the Cairo VM executes the code, certain operations take advantage of builtins to optimize logic and use less gas.
+
+[#list-of-cairo-builtins]
+.List of Cairo builtins
+
+[cols="1,2",]
+|===
+|Name of builtin | Description
+
+|Pedersen |Computes the Pedersen hash over two elements. For more information see xref:architecture_and_concepts:Cryptography/hash-functions.adoc[].
+|Poseidon |Computes the Hades permutation on three field elements. For more info, see xref to Hash functions topic. The Cairo corelib functions use this builtin under the hood. The Cairo corelib functions are defined in link:https://github.com/starkware-libs/cairo/blob/v2.6.0/corelib/src/starknet/info.cairo[`info.cairo`] in the Cairo GitHub repository.
```

Comment:
This builtin is used under the hood in [poseidon.cairo](https://github.com/starkware-libs/cairo/blob/6ee23d6bf62d4d81b9d47f67facca98c22dc4c95/corelib/src/poseidon.cairo)

Reply:
PTAL.